lib/nve/red-page.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:geocoder/geocoder.dart';
+import 'package:location/location.dart';
+import 'package:provider/provider.dart';
+import 'position-service.dart';
+import 'nve-main-page.dart';
+
+LocationData _locationData;
+bool _serviceEnabled;
+PermissionStatus _permissionGranted;
+
+class MyStatefulRedPage extends StatefulWidget {
+  MyStatefulRedPage({Key key}) : super(key: key);
+
+  @override
+  RedPage createState() => RedPage();
+}
+
+class RedPage extends State<MyStatefulRedPage> {
+  double _distance = 0;
+  String _addresse = "";
+
+  Future<void> askPermission() async {
+    _serviceEnabled = await location.serviceEnabled();
+    if (!_serviceEnabled) {
+      _serviceEnabled = await location.requestService();
+      if (!_serviceEnabled) {
+        return;
+      }
+    }
+
+    _permissionGranted = await location.hasPermission();
+    if (_permissionGranted == PermissionStatus.denied) {
+      _permissionGranted = await location.requestPermission();
+      if (_permissionGranted != PermissionStatus.granted) {
+        return;
+      }
+    }
+
+    _locationData = await location.getLocation();
+    _addresse = await getAddress();
+    if (Provider.of<PositionService>(context, listen: false).latitude != null &&
+        Provider.of<PositionService>(context, listen: false).longitude !=
+            null) {
+      _distance = await geo.distanceBetween(
+          _locationData.latitude,
+          _locationData.longitude,
+          Provider.of<PositionService>(context, listen: false).latitude,
+          Provider.of<PositionService>(context, listen: false).longitude);
+    }
+  }
+
+  Future<String> getAddress() async {
+    final _coordinates = new Coordinates(_locationData.latitude, _locationData.longitude);
+    var _addresses = await Geocoder.local.findAddressesFromCoordinates(_coordinates);
```

Comment:
this method throws exceptions when the _locationDate or _coordinates are null. Please add a check to avoir error throwing